File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/SFFunction.java
package shadow.pipeline;

import shadow.pipeline.expression.SFExpressionElement;
import shadow.pipeline.expression.SFExpressionException;
import shadow.pipeline.parameters.SFParameteri;

/**
 * @author Alessandro Martinelli
 */
public class SFFunction {

	private SFParameteri parameter;
	private SFExpressionElement function;
	
	public SFFunction(SFParameteri globalV, SFExpressionElement function) {
		super();
		this.parameter = globalV;
		if(function!=null)
			setFunction(function);
	}
	
	public SFParameteri getParameter() {
		return parameter;
	}
	
	public void setParameter(SFParameteri globalV) {
		this.parameter = globalV;
	}
	
	public SFExpressionElement getFunction() {
		return function;
	}
	
	public void compileFunction(SFParameteri[] parameters){
		this.function=this.function.cloneAsIndexed(parameters);
	}
	
	public void setFunction(SFExpressionElement function) {
		this.function = function;
		try {
			this.function.evaluateType();
			//System.err.println(function);//{*(3):[color(3), {wr2(2):{*(2):[intensity(2), tmp3(2)]}}]}
		} catch (SFExpressionException e) {
			e.printStackTrace();
		}
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/pipeline/SFFunction.js
//Java to JS on 21/03/2012

function SFFunction(globalV,func){
	this.parameter=globalV;

	if(func!=null)
		this.setFunction(func);
}

SFFunction.prototype["getParameter"]=function(){
			return this.parameter;
		};
		
SFFunction.prototype["setParameter"]=function(globalV){
			this.parameter=globalV;
		};

SFFunction.prototype["compileFunction"]=function(parameters){
			this.func=this.func.cloneAsIndexed(parameters);
		};
		
SFFunction.prototype["getFunction"]=function(){
			return this.func;
		};
		
SFFunction.prototype["setFunction"]=function(func){
			this.func = func;
			if(this.func!=undefined)
				this.func.evaluateType();
		};
		